desktop/main/autoUpdater.ts

```diff
@@ -64,26 +67,26 @@ export const checkUpdates = async (
     smappBaseDownloadUrl,
   } = currentNetwork;
   const isEqualVersion = currentVersion.compare(latestSmappRelease) === 0;
-  if (!isEqualVersion) {
-    const isOutdatedVersion = currentVersion.compare(minSmappRelease) === -1;
-    autoUpdater.allowDowngrade = true;
-    autoUpdater.autoDownload = autoDownload || isOutdatedVersion;
-    const feedUrl = getFeedUrl(smappBaseDownloadUrl, latestSmappRelease);
-    autoUpdater.setFeedURL(feedUrl);
-    try {
-      const result = await autoUpdater.checkForUpdates();
-      if (!result) return null;
-      const { updateInfo } = result;
-      return updateInfo;
-    } catch (err) {
-      if (err instanceof Error && !isNetError(err)) {
-        notifyError(mainWindow, err as Error);
-      }
-      return null;
+
+  if (isEqualVersion) {
+    return false;
+  }
+
+  const isOutdatedVersion = currentVersion.compare(minSmappRelease) === -1;
+  autoUpdater.allowDowngrade = true;
+  autoUpdater.autoDownload = autoDownload || isOutdatedVersion;
+  const feedUrl = getFeedUrl(smappBaseDownloadUrl, latestSmappRelease);
+  autoUpdater.setFeedURL(feedUrl);
+  try {
+    const result = await autoUpdater.checkForUpdates();
+    return result ? result.updateInfo : {};
+  } catch (err) {
+    if (err instanceof Error && !isNetError(err)) {
+      notifyError(mainWindow, err as Error);
     }
   }
 
-  return null;
+  return false;
```

Comment:
Previously `checkUpdates` function returns two possible values: `UpdateInfo` or `null`

Now `checkUpdates` returns three possible states: `UpdateInfo | {} | false`. Seems like a flaky anti-pattern. Let's make it clear and straightforward.


```
enum UpdateInfoStatus {
  UpdateAvailable = 0,
  UpdateNotAvailable,
  UpdateManually
}

type CheckUpdateResults =
  | { status: UpdateInfoStatus.UpdateAvailable; updateInfo: UpdateInfo; }
  | { status: UpdateInfoStatus.UpdateNotAvailable }
  | { status: UpdateInfoStatus.UpdateManually };
```